src/nlp/info.py

```diff
@@ -137,6 +137,25 @@ def _dump_license(self, file):
         """Dump license in `file` file-like object open in bytes mode (to support remote files)"""
         file.write(self.license.encode("utf-8"))
 
+    @classmethod
+    def from_merge(cls, dataset_infos: List["DatasetInfo"]):
+        dataset_infos = [dset_info for dset_info in dataset_infos if dset_info is not None]
+        description = "\n\n".join([info.description for info in dataset_infos])
+        citation = "\n\n".join([info.citation for info in dataset_infos])
+        homepage = "\n\n".join([info.homepage for info in dataset_infos])
+        license = "\n\n".join([info.license for info in dataset_infos])
+        features = None
+        supervised_keys = None
```

Comment:
For `features` and `supervised_keys` we should probably check that they're all the same, and then take the first one.

Reply:
What's the difference between a dataset schema and dataset features? I see that the schema is an underlying Arrow object, while features is a Python class, but they seem to contain the same information.

Reply:
> What's the difference between a dataset schema and dataset features?

`schema` is for the logic on the arrow side, while `features` is the logic from our side.

`nlp.Features` are simpler to create that `schema`, and allow to encode example in the correct arrow format.
`pyarrow.Schema` is used to write the arrow files.

Currently the `schema` is a public attribute of `nlp.Dataset` but we plan to hide it and only show `features` to the user. We have `nlp.Features` methods that allow to convert `features` from and to pyarrow's `schema`.

> It looks like doing a map changes the schema but not the features

Indeed x) We have an issue about it: #342 

We have to do changes in the `features` logic:
- make `features` the source of truth (and make the logic `schema` logic appear only in the `ArrowWriter`)
- make all the dataset transforms update `features` accordingly (we can infer features from the output of the given function in map for example).

> Since concatenate_datasets() will often be used in a preprocessing pipeline with many dataset transformations, comparing features is not the right behavior. It seems like since schema is the source of truth, we should rely on that and ignore the potentially incorrect features

You're right. Right now `features` can be incorrect. Until we do the changes in the `features` logic, `schema` stays the source of truth.

Let's ignore the `features` and `supervised_keys ` for now in `concatenate_datasets` then.